### PR TITLE
Kino.Input error messages

### PR DESCRIPTION
**Pull request by rhcarvalho**
Hi there!

@jonatanklosko was faster than I in addressing #325 with #326 -- thanks for the quick fix!

Since I had some changes laying around already, I decided to post them here for feedback/discussion, I wouldn't mind if they are eventually not accepted.

## ~~Tests~~

<details>

~~I noticed `Kino.Input.date` was lacking tests, and was happy to see @jonatanklosko added one yesterday!~~

~~I had a plan to cover more of the possible cases, and so tried some form of table-testing to reduce verbosity. I wonder how others see this type of test code in Elixir, given it also adds some cognitive complexity (`unquote`, `Macro.escape`, ...).~~

~~The constraints in this case are simple enough that property-based testing would also be a fun exercise :-)~~

</details>

Update: removed tests to keep the PR more focused.

## Error messages

Two out of three messages were missing part of what caused the error, making debugging unnecessarily harder.

I tried to make the messages more obvious and easier to interpret by a human when their Livebook cell is spitting out an error on them.

Something like `expected :min to be less than :max, got: #{inspect(min)} and #{inspect(max)}` is not exactly correct, as `:min` can also be *equal to* `:max`. If I were to continue the existing mathematical pattern, I'd write:

```
expected :min to be less than or equal :max, got: #{inspect(min)} and #{inspect(max)}
```

But, then, I thought why not simply state the problem directly: `:min is greater than :max`. And, when we talk about dates and time, I think the words "before" and "after" might be a better fit (might be a bias from my Go experience, where the [`time` package uses the words "before" and "after"](https://pkg.go.dev/time#Time.Compare)).

I would love to hear what are the Elixir-way ideas here, specially if there is a well establish preference.

If my changes are welcome, I could also update the other messages (time, etc) to match.

### CODE REVIEW

File: lib/kino/input.ex [hunk 1]
```diff
@@ -384,17 +384,17 @@ defmodule Kino.Input do
 
     if min && max && Date.compare(min, max) == :gt do
       raise ArgumentError,
-            "expected :min to be less than :max, got: #{inspect(min)} and #{inspect(max)}"
+            ":min = #{inspect(min)} after :max = #{inspect(max)}"
```
Comment (jonatanklosko) [inline lib/kino/input.ex:387]: It is a common phrasing in Elixir, but it's fine to reword, we should be more descriptive though:

```suggestion
            "empty date range, :min (#{inspect(min)}) is after :max (#{inspect(max)})"
```

File: lib/kino/input.ex [hunk 2]
```diff
@@ -384,17 +384,17 @@ defmodule Kino.Input do
...
     end
 
     assert_default_value!(default, "be %Date{} or nil", &(is_struct(&1, Date) or &1 == nil))
 
     if min && default && Date.compare(default, min) == :lt do
-      raise ArgumentError, "expected :default to be bigger than :min, got: #{inspect(default)}"
+      raise ArgumentError, ":default = #{inspect(default)} before :min = #{inspect(min)}"
```
Comment (jonatanklosko) [inline lib/kino/input.ex:393]:
```suggestion
      raise ArgumentError, "invalid :default, #{inspect(default)} is before :min (#{inspect(min)})"
```

File: lib/kino/input.ex [hunk 3]
```diff
@@ -384,17 +384,17 @@ defmodule Kino.Input do
...
     end
 
     if max && default && Date.compare(default, max) == :gt do
-      raise ArgumentError, "expected :default to be smaller than :max, got: #{inspect(default)}"
+      raise ArgumentError, ":default = #{inspect(default)} after :max = #{inspect(max)}"
```
Comment (jonatanklosko) [inline lib/kino/input.ex:397]:
```suggestion
      raise ArgumentError, "invalid :default, #{inspect(default)} is after :max (#{inspect(max)})"
```

File: test/kino/input_test.exs [hunk 4]
```diff
@@ -51,22 +51,61 @@ defmodule Kino.InputTest do
...
+      {[default: ~D[2023-09-01], min: ~D[2023-09-01], max: ~D[2023-09-01]],
+       %{default: ~D[2023-09-01], min: ~D[2023-09-01], max: ~D[2023-09-01]}}
+    ]
+
+    Enum.each(@valid_inputs, fn {opts, expected_attrs} ->
+      test "creates a new date input with opts #{inspect(opts)}" do
+        assert %{attrs: unquote(Macro.escape(Map.merge(expected_attrs, %{type: :date})))} =
+                 Kino.Input.date("Date Input", unquote(Macro.escape(opts)))
+      end
+    end)
```
Comment (jonatanklosko) [inline test/kino/input_test.exs:94]: There are cases where generating tests with macros makes sense, but it has its drawbacks (running a specific test, quickly identifying what failed, readability). In this case, I don't think we need the whole matrix of cases. We can test a couple edge cases with different messages, this could actually be a single test :)

**Comment by rhcarvalho** [comment]
@jonatanklosko I reviewed the existing error messages and based on your feedback updated the messages for all related functions:

- `Kino.Input.utc_datetime`
- `Kino.Input.utc_time`
- `Kino.Input.date`

Since the "expected ..." format is used in many places, I kept that + your suggestions.

I dropped the tests-related commit so that this PR can be focused on a single thing.

**Review by jonatanklosko — APPROVED** [review]
Awesome, please sync with main and it's ready to go :D

**Comment by rhcarvalho** [comment]
> please sync with main and it's ready to go :D

Done!
